### PR TITLE
[Session Replication] LPD-28291 Add liferay's tomcat session replication implementation and add functional test

### PR DESCRIPTION
**Pull request by julschong**
@tinatian 
resubmitted from https://github.com/liferay-core-infra/liferay-portal/pull/7271

per our discussion,
I split the functional test into 2 parts:
1. behavior
2. matching attributes

Matching attributes changed in this PR to use the existing clusterSampleData set in `view.jsp:94`
```
<% portletSession.setAttribute(ClusterSampleData.class.getName(), clusterSampleData); %>
```
Functional test flow:
1. localhost:8080 logged in as default test user
2. Visit portlet page and copy values on the page:
    a. cluster sample data: random string generated
    b. session Id
3.  Go to localhost:9080 portlet page to assert that the two values are the same. Meaning that the user session is the same and cluster sample data is replicated to 9080.

4. Make a new user test1
5. Go to www.able.com:8080 and log in
6. Go to portlet page and first make sure that cluster sample data and session id are different than step 2's.
7. Copy the values on the page, same as before
8. Go to www.able.com:9080 portlet page to assert that the two values are the same. Meaning that the user session is the same and cluster sample data is replicated to 9080.
9. Assert that total logged in sessions is 2

### CODE REVIEW

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/clustering/clusteringframework/ClusteringCache.testcase [hunk 1]
```diff
@@ -330,4 +267,119 @@ definition {
...
+			userEmailAddress = "test1@liferay.com");
+
+		User.firstLoginUI(
+			password = PropsUtil.get("default.admin.password"),
+			specificURL = "http://www.able.com:9080",
+			userEmailAddress = "test1@liferay.com");
+
+		Open(value1 = "http://www.able.com:9080/web/guest/session-replication-test");
+
+		Refresh();
```
Comment (julschong) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/clustering/clusteringframework/ClusteringCache.testcase:324]: Note that this refresh() is still necessary

**Comment by julschong** [comment]
ci:test:clustering

**Comment by tinatian** [comment]
ci:report:43889927

**Comment by julschong** [comment]
retest clustering, looks like maybe issue on ci

**Comment by julschong** [comment]
ci:test:clustering

**Comment by julschong** [comment]
ci:test:clustering

**Comment by julschong** [comment]
resolve conflict

**Comment by julschong** [comment]
ci:test:clustering

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:clustering

**Comment by julschong** [comment]
ci:test:clustering

**Comment by julschong** [comment]
test passed, ready for reviwe

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/clustering/clusteringframework/ClusteringCache.testcase (hunk 1 above)
Comment (tinatian) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/clustering/clusteringframework/ClusteringCache.testcase:324]: @julschong Why?

Comment (tinatian) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/clustering/clusteringframework/ClusteringCache.testcase:324]: Why you do not need refresh for the default user?

**Comment by julschong** [comment]
@tinatian will resumbit per our discussion to deal with Refresh()

**Pull request closed without merging**